django/db/migrations/questioner.py

```diff
@@ -91,9 +94,9 @@ def _boolean_input(self, question, default=None):
         return result[0].lower() == "y"
 
     def _choice_input(self, question, choices):
-        print(question)
+        self.prompt_destination.write(question)
         for i, choice in enumerate(choices):
-            print(" %s) %s" % (i + 1, choice))
+            self.prompt_destination.write(" %s) %s" % (i + 1, choice) + os.linesep)
         result = input("Select an option: ")
```

Comment:
I had a look at the [Python ticket linked above](https://bugs.python.org/issue1927), and my understanding is that most `input()` prompts go to `stderr` already, but for some edge cases where they may not, contrary to the documentation.

In a terminal on MacOS, redirecting stdout shows the input prompts not redirected (as desired for our use case):
```python
In [3]: with open('blah.txt', 'w+') as f:
   ...:     management.call_command('makemigrations', stdout=f)
   ...: 
Select an option:
```

And the content of `blah.txt` does not include the input prompt. As desired for our case.

If my understanding is correct, I wouldn't be inclined to investigate and patch the edge cases in Django; it would be better to for another interested party to patch them in CPython (as well as updating the documentation). Does that seem right to you?